bridge-token/src/lib.rs

```diff
@@ -111,6 +119,12 @@ impl BridgeToken {
     pub fn account_storage_usage(&self) -> StorageUsage {
         self.token.account_storage_usage
     }
+
+    /// Return true if the caller is either self
+    pub fn controller_or_self(&self) -> bool {
+        let caller = env::predecessor_account_id();
+        caller == self.controller || caller == env::current_account_id()
+    }
 }
```

Comment:
I remember we discussed allowing changing the metadata not only for the token account itself and factory (which is the controller) but also for some external account like `controller.factory.bridge.near`. Is there any objection to not adding it?

Reply:
To avoid adding the controller on every single bridge token, I added the factory as the "controller" and will add a function to factory that will allow making this call from a controller.

Reply:
Ok, nice!